expr/func_linearregression.go

```diff
@@ -0,0 +1,142 @@
+package expr
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"time"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/schema"
+	"github.com/raintank/dur"
+)
+
+type FuncLinearRegression struct {
+	in GraphiteFunc
+
+	startSourceAt string
+	endSourceAt   string
+
+	startTargetAt uint32
+	endTargetAt   uint32
+}
+
+func NewLinearRegression() GraphiteFunc {
+	return &FuncLinearRegression{}
+}
+
+func (s *FuncLinearRegression) Signature() ([]Arg, []Arg) {
+	return []Arg{
+			ArgSeriesList{
+				val: &s.in,
+			},
+			ArgString{
+				key:       "startSourceAt",
+				opt:       true,
+				validator: []Validator{IsATTime},
+				val:       &s.startSourceAt,
+			},
+			ArgString{
+				key:       "endSourceAt",
+				opt:       true,
+				validator: []Validator{IsATTime},
+				val:       &s.endSourceAt,
+			},
+		},
+		[]Arg{ArgSeriesList{}}
+}
+
+func (s *FuncLinearRegression) Context(context Context) Context {
+	s.startTargetAt = context.from
+	s.endTargetAt = context.to
```

Comment:
Won't you need to adjust these values to fetch more data? (similar to timeshift)